partials/KafkaConfig.java

```diff
@@ -0,0 +1,108 @@
+import org.springframework.kafka.config.KafkaListenerContainerFactory;
+import org.springframework.kafka.core.*;
+import org.springframework.kafka.listener.ConcurrentMessageListenerContainer;
+import org.springframework.kafka.support.serializer.JsonDeserializer;
+import org.springframework.kafka.support.serializer.JsonSerializer;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Configuration
+{% if hasPublish %}@EnableKafka{% endif %}
+public class Config {
+{%- if hasSubscribe or hasPublish %}
+    @Value("${kafka.bootstrap-servers:localhost:9092}")
+    private String bootstrapServers;
+{% endif %}
+{%- if hasPublish %}
+    @Value("${kafka.subscribe.pool-timeout:3000}")
+    private long poolTimeout;
+
+    @Value("${kafka.subscribe.amount-of-listeners:3}")
+    private Integer amountOfListeners;
+{% endif %}
+{%- if hasSubscribe %}
+    @Bean
+    public KafkaTemplate<Integer, String> kafkaTemplate() {
+        return new KafkaTemplate<>(producerFactory());
+    }
+
+    @Bean
+    public ProducerFactory<Integer, String> producerFactory() {
+        return new DefaultKafkaProducerFactory<>(producerConfigs());
+    }
+
+    @Bean
+    public Map<String, Object> producerConfigs() {
+        Map<String, Object> props = new HashMap<>();
+        props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+        props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, IntegerSerializer.class);
+        props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, JsonSerializer.class);
+        props.put(JsonSerializer.TYPE_MAPPINGS,
+    {%- for schema in asyncapi.allSchemas().values() %}
+        {%- if schema.uid() | first !== '<' %}
+        "{{schema.uid()}}:com.asyncapi.model.{{schema.uid() | camelCase | upperFirst}}{% if not loop.last %}," +{% else %}"{% endif %}
+        {% endif -%}
+    {% endfor -%}
+        );
+        // See https://kafka.apache.org/documentation/#producerconfigs for more properties
+        return props;
+    }
+{% endif %}
+{%- if hasPublish %}
+    @Bean
+    KafkaListenerContainerFactory<ConcurrentMessageListenerContainer<Integer, String>>
+    kafkaListenerContainerFactory() {
+        ConcurrentKafkaListenerContainerFactory<Integer, String> factory =
+                new ConcurrentKafkaListenerContainerFactory<>();
+        factory.setConsumerFactory(consumerFactory());
+        factory.setConcurrency(amountOfListeners);
+        factory.getContainerProperties().setPollTimeout(poolTimeout);
+        return factory;
+    }
+
+    @Bean
+    public ConsumerFactory<Integer, String> consumerFactory() {
+        return new DefaultKafkaConsumerFactory<>(consumerConfigs());
+    }
+
+    @Bean
+    public Map<String, Object> consumerConfigs() {
+        Map<String, Object> props = new HashMap<>();
+        props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+        props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
```

Comment:
Why do we use String here?

Reply:
i'll change to Integer for seek of unification